tools/rosgraph/src/rosgraph/roslogging.py

```diff
@@ -39,12 +39,31 @@
 import time
 import logging
 import logging.config
+import inspect
 
 import rospkg
 from rospkg.environment import ROS_LOG_DIR
 
 class LoggingException(Exception): pass
 
+class RospyLogger(logging.getLoggerClass()):
+    def findCaller(self):
+        """
+        Find the stack frame of the caller so that we can note the source
+        file name, line number, and function name with class name if possible.
+        """
+        frame, _, lineno, _, _, _ = inspect.stack()[3]
+        file_name = inspect.getabsfile(frame)
+        func_name = frame.f_code.co_name
+        try:
+            class_name = frame.f_locals['self'].__class__.__name__
+            func_name = '%s.%s' % (class_name, func_name)
+        except KeyError:
+            pass
+        return file_name, lineno, func_name
```

Comment:
I think it would be good to call the implementation of the base class here and then only "enhance" the function name with additional information.